src/deluge/gui/menu_item/arpeggiator/ratchet_amount.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2014-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include "gui/menu_item/unpatched_param.h"
+#include "gui/ui/sound_editor.h"
+
+namespace deluge::gui::menu_item::arpeggiator {
+class RatchetAmount : public UnpatchedParam {
+public:
+	using UnpatchedParam::UnpatchedParam;
+	bool isRelevant(ModControllableAudio* modControllable, int32_t whichThing) override {
+		return !soundEditor.editingKit() && !soundEditor.editingCVOrMIDIClip();
```

Comment:
```suggestion
		return getCurrentOutputType() != OutputType::AUDIO;
```

We do support ARP of kit rows, CV, and MIDI outputs I think? Or is there specific limitations on the ratchet implementation that prevents it working there?

Reply:
Right now ratchets are limited to internal synth engine. The arp gates are calculated differently and I need to refactor it, but better in a future PR. Right now it works smoothly for synth clips only. Kits and Midi are a future improvement (maybe during the beta phase these weeks I will be able to add them, but want the PR out so people can start using/testing it)